var/spack/repos/builtin/packages/quantlib/package.py

```diff
@@ -0,0 +1,159 @@
+
+        if "+test" in self.spec:
+            args += ["-DQL_BUILD_TEST_SUITE=ON", "-DQL_INSTALL_TEST_SUITE=ON"]
+        else:
+            args += ["-DQL_BUILD_TEST_SUITE=OFF", "-DQL_INSTALL_TEST_SUITE=OFF"]
+
+        if "+openmp" in self.spec:
+            args += ["-DQL_ENABLE_OPENMP=ON"]
+        else:
+            args += ["-DQL_ENABLE_OPENMP=OFF"]
+
+        if "+punit" in self.spec:
+            args += ["-DQL_ENABLE_PARALLEL_UNIT_TEST_RUNNER=ON"]
+        else:
+            args += ["-DQL_ENABLE_PARALLEL_UNIT_TEST_RUNNER=OFF"]
+
+        if "+sessions" in self.spec:
+            args += ["-DQL_ENABLE_SESSIONS=ON"]
+        else:
+            args += ["-DQL_ENABLE_SESSIONS=OFF"]
+
+        if "+threadsafe" in self.spec:
+            args += ["-DQL_ENABLE_THREAD_SAFE_OBSERVER_PATTERN=ON"]
+        else:
+            args += ["-DQL_ENABLE_THREAD_SAFE_OBSERVER_PATTERN=OFF"]
+
+        if "+tracing" in self.spec:
+            args += ["-DQL_ENABLE_TRACING=ON"]
+        else:
+            args += ["-DQL_ENABLE_TRACING=OFF"]
+
+        if "+errorfunctions" in self.spec:
+            args += ["-DQL_ERROR_FUNCTIONS=ON"]
+        else:
+            args += ["-DQL_ERROR_FUNCTIONS=OFF"]
+
+        if "+errorlines" in self.spec:
+            args += ["-DQL_ERROR_LINES=ON"]
+        else:
+            args += ["-DQL_ERROR_LINES=OFF"]
+
+        if "+extra" in self.spec:
+            args += ["-DQL_EXTRA_SAFETY_CHECKS=ON"]
+        else:
+            args += ["-DQL_EXTRA_SAFETY_CHECKS=OFF"]
+
+        if "+high" in self.spec:
+            args += ["-DQL_HIGH_RESOLUTION_DATE=ON"]
+        else:
+            args += ["-DQL_HIGH_RESOLUTION_DATE=OFF"]
+
+        if "+null" in self.spec:
+            args += ["-DQL_NULL_AS_FUNCTIONS=ON"]
+        else:
+            args += ["-DQL_NULL_AS_FUNCTIONS=OFF"]
+
+        if "+tagged" in self.spec:
+            args += ["-DQL_TAGGED_LAYOUT=ON"]
+        else:
+            args += ["-DQL_TAGGED_LAYOUT=OFF"]
+
+        if "+tidy" in self.spec:
+            args += ["-DQL_USE_CLANG_TIDY=ON"]
+        else:
+            args += ["-DQL_USE_CLANG_TIDY=OFF"]
+
+        if "+indexed" in self.spec:
+            args += ["-DQL_USE_INDEXED_COUPON=ON"]
+        else:
+            args += ["-DQL_USE_INDEXED_COUPON=OFF"]
+
+        if "+std" in self.spec:
+            args += ["-DQL_USE_STD_CLASSES=ON"]
+        else:
+            args += ["-DQL_USE_STD_CLASSES=OFF"]
+
+        return args
+
+    # The Quantlib build process creates the `quantlib-config` file but it is
+    # not part of the paths that get installed via spack.  This function updates
+    # that file to include the spack paths needed and then copies the file from the
+    # build directory to the prefix.bin directory.
+    @run_after("cmake")
+    def install_quantlib_config(self):
+        import os
+
+        prefix = self.spec.prefix
+        mkdir = which("mkdir")
+        mkdir("-p", prefix.bin)
```

Comment:
```suggestion
        mkdirp(prefix.bin)
```
Ref: https://spack.readthedocs.io/en/latest/llnl.util.html#llnl.util.filesystem.mkdirp.